http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -1478,41 +756,181 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
           AppControlledResourceStatus.Created
         )
         .transaction
-    } yield dbName
+    } yield result
   }
 
-  private[util] def deleteAppWsmResources(app: App, workspaceId: WorkspaceId)(implicit
+  private[util] def createWsmKubernetesNamespaceResource(app: App,
+                                                         workspaceId: WorkspaceId,
+                                                         namespacePrefix: String,
+                                                         databases: List[UUID],
+                                                         identity: Option[UUID]
+  )(implicit ev: Ask[F, AppContext]): F[CreatedControlledAzureKubernetesNamespaceResult] =
+    for {
+      ctx <- ev.ask
+      _ <- logger.info(ctx.loggingCtx)(
+        s"Creating $namespacePrefix namespace for app ${app.appName.value} in cloud workspace ${workspaceId.value}"
+      )
+
+      // Build WSM clients
+      wsmApi <- buildWsmControlledResourceApiClient
+      wsmResourceApi <- buildWsmResourceApiClient
+
+      // Name of the WSM resource. Must be unique per workspace.
+      // For shared apps, name it by the appType so it's semantically meaningful.
```

Comment:
Nit that may not need to be addressed ever, but with the change implemented by AOU, appType is not really unique to a chart now, would this eventually break the assumption made here?

Reply:
Yeah I guess so. :-/ Kind of academic I guess though, since `ALLOWED` cannot be launched in shared mode.

Reply:
Oh very true, we might get shared mode there in the future but for now feel free to ignore my nit picking 😬